### PR TITLE
Feature/minor improvments

### PR DESCRIPTION
**Pull request by phimichel**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+\.idea/
+
+*.iml
+target/
```
Comment (pthum) [inline .gitignore]: would be better to have the "extended" gitignore (with bin, target, .DS_Store, .project, .settings, etc.) in here

File: src/main/java/com/apiomat/helper/mvnnmhelper/mojos/UploadNMMojo.java [hunk 2]
```diff
@@ -63,17 +57,17 @@
...
 	 */
-	@Parameter( defaultValue = "${project.build.finalName}", readonly = true )
+	@Parameter( defaultValue = "${project.build.finalName}")
 	private String finalName;
 
 	@Override
 	public void executeRequest( ) throws MojoExecutionException, IOException
 	{
 
-		final URI hostUrl = buildHostUrl( );
```
Comment (pthum) [inline src/main/java/com/apiomat/helper/mvnnmhelper/mojos/UploadNMMojo.java]: why did you remove all the finals in this file?

File: src/main/java/com/apiomat/helper/mvnnmhelper/mojos/UploadNMMojo.java [hunk 3]
```diff
@@ -10,24 +10,18 @@
...
-import java.net.URL;
-
 import org.apache.http.client.utils.URIBuilder;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.Parameter;
 import org.apache.maven.shared.utils.StringUtils;
 
+import java.io.*;
```
Comment (pthum) [inline src/main/java/com/apiomat/helper/mvnnmhelper/mojos/UploadNMMojo.java]: we don't use such general imports

File: src/main/java/com/apiomat/helper/mvnnmhelper/mojos/UpdateVersionNMMojo.java [hunk 4]
```diff
@@ -10,16 +10,15 @@
...
+import com.apiomat.helper.mvnnmhelper.VersionCompareHelper;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.Parameter;
 import org.codehaus.plexus.util.StringUtils;
 
-import com.apiomat.helper.mvnnmhelper.ModuleUpdateManager;
-import com.apiomat.helper.mvnnmhelper.VersionCompareHelper;
+import java.io.File;
```
Comment (pthum) [inline src/main/java/com/apiomat/helper/mvnnmhelper/mojos/UpdateVersionNMMojo.java]: would be better if such changes on the import order are reverted before a commit. can you change it back? (also affects the other files)

File: src/main/java/com/apiomat/helper/mvnnmhelper/mojos/PackageNMMojo.java [hunk 5]
```diff
@@ -115,7 +124,7 @@ private Object getFieldValue( final String fieldName )
 			f.setAccessible( true );
 			return f.get( this );
 		}
-		catch ( NoSuchFieldException | SecurityException | IllegalArgumentException | IllegalAccessException e )
+		catch ( final NoSuchFieldException | SecurityException | IllegalArgumentException | IllegalAccessException e )
```
Comment (pthum) [inline src/main/java/com/apiomat/helper/mvnnmhelper/mojos/PackageNMMojo.java]: final removed?

File: src/main/java/com/apiomat/helper/mvnnmhelper/mojos/AbstractRequestMojo.java [hunk 6]
```diff
@@ -109,18 +95,24 @@
 		}
 
 		@Override
-		public void checkClientTrusted( final X509Certificate[ ] certs, final String authType )
+		public void checkClientTrusted( final X509Certificate[] certs, final String authType )
 		{}
 
 		@Override
-		public void checkServerTrusted( final X509Certificate[ ] certs, final String authType )
+		public void checkServerTrusted( final X509Certificate[] certs, final String authType )
```
Comment (pthum) [inline src/main/java/com/apiomat/helper/mvnnmhelper/mojos/AbstractRequestMojo.java]: whitespaces removed

File: src/main/java/com/apiomat/helper/mvnnmhelper/mojos/AbstractRequestMojo.java [hunk 7]
```diff
@@ -210,6 +203,45 @@ protected void executeGoal( final String goal ) throws MojoExecutionException
...
+		final Set<String> parameterNames )
+	{
+		Xpp3Dom result = null;
+		if ( config != null )
+		{
+			final int childCount = config.getChildCount( );
+			result = new Xpp3Dom( config.getName( ) );
+			result.setValue( config.getValue( ) );
+
+			final String[] attributeNames = config.getAttributeNames( );
```
Comment (pthum) [inline src/main/java/com/apiomat/helper/mvnnmhelper/mojos/AbstractRequestMojo.java]: is there a case where getAttributeNames may return null?

File: src/main/java/com/apiomat/helper/mvnnmhelper/mojos/PackageNMMojo.java (hunk 5 above)
Comment (phimichel) [reply src/main/java/com/apiomat/helper/mvnnmhelper/mojos/PackageNMMojo.java]: final added or?

File: src/main/java/com/apiomat/helper/mvnnmhelper/mojos/AbstractRequestMojo.java (hunk 6 above)
Comment (phimichel) [reply src/main/java/com/apiomat/helper/mvnnmhelper/mojos/AbstractRequestMojo.java]: It's against our coding guidelines this whitespace ;)

File: src/main/java/com/apiomat/helper/mvnnmhelper/mojos/AbstractRequestMojo.java (hunk 7 above)
Comment (phimichel) [reply src/main/java/com/apiomat/helper/mvnnmhelper/mojos/AbstractRequestMojo.java]: I don't thinks so but I will add a null check

File: src/main/java/com/apiomat/helper/mvnnmhelper/mojos/UpdateVersionNMMojo.java (hunk 4 above)
Comment (phimichel) [reply src/main/java/com/apiomat/helper/mvnnmhelper/mojos/UpdateVersionNMMojo.java]: No sorry I haven't the chance to revert this. That will be a lot of work

**Comment by phimichel** [comment]
@pthum Ready for reviewing comments and code again
